packages/contracts/src/zksync/TokenVotingSetupZkSync.sol

```diff
@@ -186,6 +214,18 @@ contract TokenVotingSetup is PluginSetup {
             );
         }
 
+        if (setUpgradePermission) {
+            bytes32 tokenUpgradePermission = GovernanceERC20Upgradeable(token)
+                .UPGRADE_GOVERNANCE_ERC20_PERMISSION_ID();
+            permissions[permissions.length - 1] = PermissionLib.MultiTargetPermission(
```

Comment:
Please think carefully about this: I believe the `UPGRADE` permission will always be the final permission in the array if it is needed, but we won't know if it is at index 4 or 5 at compile time. This depends on the token being address(0) or not.

Can this break? Can this be exploited?